pkg/skaffold/runner/listen.go

```diff
@@ -60,29 +62,45 @@ func (l *SkaffoldListener) WatchForChanges(ctx context.Context, out io.Writer, d
 
 	l.LogWatchToUser(out)
 
+	eg, egCtx := errgroup.WithContext(ctx)
+	devCtx, cancelDev := context.WithCancel(egCtx)
 	for {
 		select {
-		case <-ctx.Done():
+		case <-egCtx.Done():
+			cancelDev()
 			return nil
 		case <-l.intentChan:
-			if err := l.do(devLoop); err != nil {
+			// TODO(dgageot): make sure a dev cycle triggered by an intent can be interrupted.
+			if err := l.do(devCtx, devLoop); err != nil {
```

Comment:
Why are you calling this directly instead of through `eg.Go()` like you do below?

Reply:
If I remember correctly, it didn't work just by using the same code than under `<-trigger`. So I decided to just add a TODO for later.